apollo-server/v3/integration-testing/src/server.ts

```diff
@@ -0,0 +1,34 @@
+import { ApolloServer, gql } from 'apollo-server';
+
+// Schema definition
+export const typeDefs = gql`
+  type Query {
+    hello(name: String): String!
+  }
+`;
+
+// Resolver map
+export const resolvers = {
+  Query: {
+    hello: (_, { name }) => `Hello ${name}!`,
+  },
+};
+
+export const createApolloServer = async (options = { port: 4000 }) => {
+  const server = new ApolloServer({
+    typeDefs,
+    resolvers,
+  });
+
+  return server
+    .listen(options, () => {
+      if (process.env.NODE_ENV !== 'test') {
+        console.log(
+          `🚀 Query endpoint ready at http://localhost:${options.port}${server.graphqlPath}`,
+        );
+      }
+    })
+    .then((serverResponse) => {
```

Comment:
This `then` doesn't do anything: it takes a Promise that returns serverResponse, and returns ... a promise that returns serverResponse.

(And in general, if you're in an async function you probably want to just use `await` which lets you use the normal JS language-level features of "write one thing after another" and "try/catch blocks" instead of `.then` and `.catch`.)